core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolutionHelperService.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for solution editor.
+ */
+
+oppia.constant('SUPPORTED_INTERACTIONS', [
+    'InteractiveMap',
+    'MusicNotesInput',
+    'GraphInput',
+    'SetInput',
+    'MathExpressionInput',
+    'MultipleChoiceInput',
+    'ImageClickInput',
+    'ItemSelectionInput'
+]);
+
+oppia.constant('UNSUPPORTED_INTERACTION_OBJECT_TYPES', {
+  CodeRepl: 'CodeString',
+  PencilCodeEditor: 'CodeString',
+  NumericInput: 'Real',
+  TextInput: 'NormalizedString',
+  LogicProof: 'LogicQuestion'
+});
+
+oppia.factory('StateSolutionHelperService', [
+  'oppiaHtmlEscaper', 'responsesService', 'stateInteractionIdService',
+  'SUPPORTED_INTERACTIONS', 'UNSUPPORTED_INTERACTION_OBJECT_TYPES',
+  function(oppiaHtmlEscaper, responsesService, stateInteractionIdService,
+    SUPPORTED_INTERACTIONS, UNSUPPORTED_INTERACTION_OBJECT_TYPES) {
+    return {
+      isSupportedInteraction: function (id) {
+        return (SUPPORTED_INTERACTIONS.indexOf(id) === -1);
+      },
+      getUnsupportedInteractionObjectType: function (id) {
+        return UNSUPPORTED_INTERACTION_OBJECT_TYPES[id];
+      },
+      getCorrectAnswer: function (solution) {
```

Comment:
This is better defined as a method on `solution` -- something like solution.getCorrectAnswer(interactionId).

Actually, thinking about it, a solution is always associated with an interactionId, and knowing the interactionId will be helpful when processing the solution, right? So perhaps you should include that as part of the frontend Solution object when constructing it. (It does not have to be part of the backend one.)